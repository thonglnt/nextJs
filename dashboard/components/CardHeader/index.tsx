import { memo } from 'react'
import { Box, Typography, TypographyProps, useTheme } from '@mui/material'
import MoreVertIcon from '@mui/icons-material/MoreVert'

// Components
import Button from '@components/Button'

export interface ICardHeader extends TypographyProps {
  title: string
  desc?: string
  actions?: boolean
}

const CardHeader = ({ title, desc, actions = false }: ICardHeader) => {
  const theme = useTheme()

  return (
    <Box display="flex" alignItems="center" width="100%">
      <Box>
        <Typography
          variant="h5"
          fontSize="18px"
          fontWeight={700}
          lineHeight="25px"
        >
          {title}
        </Typography>
        {desc && (
          <Typography
            fontSize="14px"
            lineHeight="21px"
            color={theme.palette.grey[500]}
          >
            {desc}
          </Typography>
        )}
      </Box>
      {actions && (
        <Button>
          <MoreVertIcon />
        </Button>
      )}
    </Box>
  )
}

export default memo(CardHeader)
