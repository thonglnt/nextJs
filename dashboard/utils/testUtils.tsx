import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
