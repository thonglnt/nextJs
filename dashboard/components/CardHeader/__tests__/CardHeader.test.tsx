// Utils
import { render, screen } from '@utils/testUtils'

// Components
import CardHeader, { ICardHeader } from '..'

describe('Card Header component', () => {
  const mockProps: ICardHeader = {
    title: 'Learn more',
    desc: 'description',
    actions: true,
  }

  it('should render Card Header component', () => {
    const { container } = render(<CardHeader {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should render match content Card Header component', () => {
    render(<CardHeader {...mockProps} />)

    const contentCardHeader = screen.getByText(/Learn more/i)
    const descCardHeader = screen.getByText(/description/i)

    expect(contentCardHeader).toHaveTextContent('Learn more')
    expect(descCardHeader).toHaveTextContent('description')
  })
})
