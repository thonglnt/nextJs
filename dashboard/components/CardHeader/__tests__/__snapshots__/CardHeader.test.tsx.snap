// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Header component should render Card Header component 1`] = `
<div>
  <div
    class="MuiBox-root css-7pf6at"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
      >
        Learn more
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
      >
        description
      </p>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-oqx3s-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MoreVertIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
